### PR TITLE
Add RearrangeOp

### PR DESCRIPTION
**Pull request by fzimmermann89**

### CODE REVIEW

File: src/mrpro/operators/RearrangeOp.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        Parameters
+        ----------
+        pattern
+            Pattern describing the forward of the operator.
+            Also see einops.rearrange for more information.
+            Example: "... h w -> ... (w h)"
+        additional_info
+            Additional information passed the the rearrange function,
+            describing the the size of certain dimensions.
```
Comment (ckolbPTB) [inline src/mrpro/operators/RearrangeOp.py]: one the too many

File: src/mrpro/operators/RearrangeOp.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        """Initialize Einsum Operator.
+
+        Parameters
+        ----------
+        pattern
+            Pattern describing the forward of the operator.
+            Also see einops.rearrange for more information.
+            Example: "... h w -> ... (w h)"
+        additional_info
+            Additional information passed the the rearrange function,
```
Comment (ckolbPTB) [inline src/mrpro/operators/RearrangeOp.py]: the -> to

**Review by ckolbPTB — APPROVED** [review]

**Comment by ckolbPTB** [comment]
Maybe consider using the same terms as in einops.rearrange self._adjoint/forward_rule  -> self._adjoint/forward_pattern

**Comment by fzimmermann89** [comment]
> Maybe consider using the same terms as in einops.rearrange self._adjoint/forward_rule -> self._adjoint/forward_pattern

I also changed in in einsumop, which used the same names.

**Comment by fzimmermann89** [comment]
pre-commit.ci autofix

**Pull request merged**